frontend/src/contexts/HostContext/HostContext.js

```diff
@@ -108,28 +108,48 @@ function HostProvider({ children }) {
 
   useEffect(() => {
     if (state.newPlayerStaging.length) {
-      const newPlayerIDs = state.newPlayerStaging.map(
-        (player) => player.playerId,
-      );
-
-      const message =
-        state.gameState === 'waiting-for-players'
-          ? {
-              big: "You've joined the lobby",
-              small: 'Please wait for the host to start the game',
-            }
-          : {
-              big: 'A round is in progress',
-              small: 'You will join the next round automatically',
-            };
-
-      dispatch({
-        type: 'SEND_PLAYER_CONNECTED_MESSAGES',
-        payload: {
-          players: newPlayerIDs,
-          message,
-        },
-      });
+      const numberOfPlayersExceedingLimit =
+        state.playerIDs.length +
+        state.newPlayerStaging.length -
+        state.gameSettings.maxPlayers;
+
+      if (numberOfPlayersExceedingLimit > 0) {
+        const playersExceedingLimit = state.newPlayerStaging.slice(
+          -numberOfPlayersExceedingLimit,
+        );
+
+        playersExceedingLimit.forEach((player) => {
+          dispatch({
+            type: 'TOO_MANY_PLAYERS',
+            payload: {
+              player: player.playerId,
+            },
+          });
+        });
```

Comment:
I think this would be cleaner if instead of sending a message for each player it sent one message with all of the offending players.